### PR TITLE
[C++ bridge] Add support for move constructor of std::vector.

### PR DESCRIPTION
**Pull request by schweitzpgi**
Allow the use of the move constructor so that cases such as

	std::vector<int> vectorValue(N);
	...
	return vectorValue;

will compile and simple vectors of arithmetic types may be used.  Note that this requires the CUDA-Q runtime to actually return these values to the caller rather than dropping them on the floor after a launched kernel has deactivated.  That is not always the case, however, since some launchers do not expect any return value(s).

### CODE REVIEW

**Review by amccaskey — APPROVED** [review]
tested locally, lgtm

**Comment by schweitzpgi** [comment]
Jobs took so long, the cache was purged out from underneath them.

File: lib/Optimizer/Transforms/GenKernelExecution.cpp [hunk 1]
```diff
@@ -1776,11 +1778,47 @@ class GenerateKernelExecution
...
+            auto sv = builder.create<cudaq::cc::StdvecInitOp>(loc, ty, buffPtr,
+                                                              buffSize);
+            call.getResult(0).replaceAllUsesWith(sv);
+            deadCalls.push_back(call);
+          }
+        }
+      });
+      for (auto *op : deadCalls)
+        op->erase();
+      module.dump();
```
Comment (bmhowe23) [inline lib/Optimizer/Transforms/GenKernelExecution.cpp:1814]: @schweitzpgi - is this intentional?

**Pull request merged**